src/twin/mod.rs

```diff
@@ -91,43 +94,58 @@ pub struct Twin {
 }
 
 impl Twin {
-    pub fn new(client: Box<dyn IotHub>) -> Self {
+    pub async fn new(client: Box<dyn IotHub>) -> Self {
         let (tx_reported_properties, rx_reported_properties) = mpsc::channel(100);
         let (tx_outgoing_message, rx_outgoing_message) = mpsc::channel(100);
 
+        let mut features = HashMap::from([
```

Comment:
as discussed I propose to move initialization to `Feature::report_initial_state`